bodhi/client/__init__.py

```diff
@@ -43,6 +43,48 @@ def _warn_if_url_and_staging_set(ctx, param, value):
     return value
 
 
+def _validate_query_user(ctx, param, value):
+    """
+    Callback used on the --user flag on query commands (query overrides and query updates)
+    to check if both --user and --mine are set, and return a warning message to stderr
+
+    Args:
+        ctx (click.core.Context): The Click context, used to find out if the --mine flag is set.
+        param (click.core.Option): The option being handled. Unused.
+        value (unicode): The value of the --user flag.
+
+    Returns:
+        unicode: The value of the --user flag.
+    """
+    if value and ctx.params.get('mine'):
+        click.echo('Warning: --mine and --user options are both set. --user will be ignored.\n',
+                   err=True)
+    return value
+
+
+def _validate_query_mine(ctx, param, value):
+    """
+    Callback used on the --mine flag on query commands (query overrides and query updates)
+    to check if the USERNAME environment variable is set. If it isn't set, raise a
+    click.BadParameter exception.
+
+    Args:
+        ctx (click.core.Context): The Click context, used to find out if the --user flag is set.
+        param (click.core.Option): The option being handled. Unused.
+        value (Boolean): True if the --mine flag was given.
+
+    Returns:
+        Boolean: The value of the --mine flag.
+    """
+    if value and not os.environ['USERNAME']:
+        raise click.BadParameter('USERNAME environment variable not set.')
```

Comment:
Bit confused with this here. I like the init_username idea, but with the current check here in place, the validator won't pass, and we wont get down to running the init_username.

IMHO, we could just use the init_username for the --mine, and not worry about the environment variable at all. This means that a user that has logged in before, --mine will just work, otherwise, they will need to type in their username every time --mine is run.

Of course, once we enable openidconnect, this will make a little more sense i think.